### ISSUE TITLE
XCode 13, iOS 15 lots of reports of blocked CFPREFERENCES

### ISSUE THREAD

**Issue by foxware00** [firebase/firebase-ios-sdk#8710]
### Step 1: Describe your environment

  * Xcode version: 13
  * Firebase SDK version: 8.7.0
  * Installation method: `CocoaPods 1.11.2`
  * Firebase Component: _____ (Core, Crashlytics, DynamicLinks)

### Step 2: Describe the problem

Getting a lot of crash reports and all of them have thread 0 in the same state. Blocked on `CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD` I can only assume something is funky in iOS 15 that is causing these crashes. See frame `8`

#### Steps to reproduce:

Unfornately unable to repro, happens in the wild.

#### Relevant Code:

```
Thread 0 name:
Thread 0:
0   libsystem_kernel.dylib        	0x00000001b6f79564 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001b6f79bfc mach_msg + 76 (mach_msg.c:119)
2   libdispatch.dylib             	0x00000001803b728c _dispatch_mach_send_and_wait_for_reply + 520 (mach.c:815)
3   libdispatch.dylib             	0x00000001803b763c dispatch_mach_send_with_result_and_wait_for_reply + 56 (mach.c:2019)
4   libxpc.dylib                  	0x00000001f048ab9c xpc_connection_send_message_with_reply_sync + 240 (connection.c:974)
5   CoreFoundation                	0x0000000180759c20 __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke_2 + 40 (CFPrefsSearchListSource.m:1025)
6   CoreFoundation                	0x00000001807266f8 -[_CFXPreferences withConnectionForRole:performBlock:] + 52 (CFXPreferences.m:1428)
7   CoreFoundation                	0x000000018074941c __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke + 140 (CFPrefsSearchListSource.m:1024)
8   CoreFoundation                	0x0000000180761cb0 CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD + 100 (CFPrefsSearchListSource.m:947)
9   CoreFoundation                	0x00000001807804ec -[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:] + 332 (CFPrefsSearchListSource.m:1044)
10  CoreFoundation                	0x00000001806d7824 -[CFPrefsSearchListSource alreadylocked_generationCountFromListOfSources:count:] + 232 (CFPrefsSearchListSource.m:1077)
11  CoreFoundation                	0x00000001806c6d04 -[CFPrefsSearchListSource alreadylocked_getDictionary:] + 468 (CFPrefsSearchListSource.m:1175)
12  CoreFoundation                	0x000000018072755c -[CFPrefsSearchListSource alreadylocked_copyValueForKey:] + 172 (CFPrefsSearchListSource.m:614)
13  CoreFoundation                	0x000000018078e958 -[CFPrefsSource copyValueForKey:] + 60 (CFPrefsSource.m:876)
14  CoreFoundation                	0x00000001806c8d7c __76-[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:]_block_invoke + 44 (CFXPreferences.m:1014)
15  CoreFoundation                	0x00000001806e8308 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 404 (CFPrefsSearchListSource.m:1728)
16  CoreFoundation                	0x00000001806c0328 normalizeQuintuplet + 356 (CFPrefsSearchListSource.m:74)
17  CoreFoundation                	0x00000001806b93ac -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 152 (CFPrefsSearchListSource.m:1600)
18  CoreFoundation                	0x00000001807170ec -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 168 (CFXPreferences.m:1010)
19  CoreFoundation                	0x00000001806dd4d8 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 128 (CFXPreferences.m:1896)
20  AppKit                      	0x00000001068b39bc 0x105b6c000 + 13924796
21  AppKit                      	0x00000001068109d0 +[APMAnalytics persistedConsentSource] + 40
22  AppKit                      	0x0000000106810814 +[APMAnalytics loadPersistedConsentSettings] + 32
23  AppKit                      	0x000000010680f690 __46+[APMAnalytics startWithAppID:origin:options:]_block_invoke + 80
24  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
25  libdispatch.dylib             	0x000000018039dee4 _dispatch_once_callout + 32 (once.c:52)
26  AppKit                      	0x000000010680f634 +[APMAnalytics startWithAppID:origin:options:] + 1048
27  AppKit                      	0x000000010680a230 +[FIRAnalytics startWithConfiguration:options:] + 532
28  AppKit                      	0x00000001062036f4 -[FIRApp configureCore] + 256 (FIRApp.m:368)
29  AppKit                      	0x00000001062033e0 +[FIRApp addAppToAppDictionary:] + 100 (FIRApp.m:324)
30  AppKit                      	0x0000000106202b7c +[FIRApp configureWithName:options:] + 704 (FIRApp.m:208)
31  AppKit                      	0x00000001062027f8 +[FIRApp configureWithOptions:] + 96 (FIRApp.m:151)
32  AppKit                      	0x0000000106202784 +[FIRApp configure] + 124 (FIRApp.m:137)
33  App                         	0x0000000104e60648 specialized AppDelegate.application(_:didFinishLaunchingWithOptions:) + 436 (AppDelegate.swift:41)
34  App                         	0x0000000104e591a8 application + 4 (<compiler-generated>:0)
35  App                         	0x0000000104e591a8 @objc AppDelegate.application(_:didFinishLaunchingWithOptions:) + 132
36  UIKitCore                     	0x0000000182e5c1ac -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 356 (UIApplication.m:2329)
37  UIKitCore                     	0x0000000183044548 -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 3524 (UIApplication.m:2741)
38  UIKitCore                     	0x000000018302ceb4 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1208 (UIApplication.m:4474)
39  UIKitCore                     	0x0000000182e8a054 -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 152 (_UISceneLifecycleMultiplexer.m:452)
40  UIKitCore                     	0x0000000182ce16a4 _UIScenePerformActionsWithLifecycleActionMask + 104 (_UISceneLifecycleState.m:109)
41  UIKitCore                     	0x0000000182e0db78 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 224 (_UISceneLifecycleMultiplexer.m:565)
42  UIKitCore                     	0x0000000182f2e644 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 380 (_UISceneLifecycleMultiplexer.m:517)
43  UIKitCore                     	0x000000018316f7bc -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 760 (_UISceneLifecycleMultiplexer.m:564)
44  UIKitCore                     	0x00000001830faed4 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 340 (_UISceneLifecycleMultiplexer.m:470)
45  UIKitCore                     	0x0000000182ce24a8 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block... + 196 (_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction.m:73)
46  UIKitCore                     	0x0000000182da2c88 +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:actions:completion:] + 892 (BSAnimationSettings+UIKit.m:50)
47  UIKitCore                     	0x0000000182ce43b8 _UISceneSettingsDiffActionPerformChangesWithTransitionContext + 276 (_UISceneSettingsDiffAction.m:23)
48  UIKitCore                     	0x0000000182dcf7e8 -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 384 (_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction.m:58)
49  UIKitCore                     	0x00000001832577bc __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke.610 + 776 (UIScene.m:1775)
50  UIKitCore                     	0x0000000182d131a4 -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 256 (UIScene.m:1476)
51  UIKitCore                     	0x0000000182d5693c -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 288 (UIScene.m:1742)
52  UIKitCore                     	0x0000000182e8ef18 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 572 (UIApplication.m:3937)
53  UIKitCore                     	0x0000000182dcdaa8 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 388 (UIApplicationSceneClientAgent.m:45)
54  FrontBoardServices            	0x00000001927afe30 -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 440 (FBSScene.m:439)
55  FrontBoardServices            	0x00000001927d5cec __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke.215 + 128 (FBSWorkspaceScenesClient.m:384)
56  FrontBoardServices            	0x00000001927906c4 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240 (FBSWorkspace.m:352)
57  FrontBoardServices            	0x0000000192791d04 __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke + 372 (FBSWorkspaceScenesClient.m:383)
58  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
59  libdispatch.dylib             	0x00000001803a00f4 _dispatch_block_invoke_direct + 264 (queue.c:489)
60  FrontBoardServices            	0x0000000192791fa4 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48 (FBSSerialQueue.m:157)
61  FrontBoardServices            	0x00000001927913e4 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220 (FBSSerialQueue.m:181)
62  FrontBoardServices            	0x00000001927959f4 -[FBSSerialQueue _performNextFromRunLoopSource] + 28 (FBSSerialQueue.m:194)
63  CoreFoundation                	0x000000018074c020 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28 (CFRunLoop.c:1972)
64  CoreFoundation                	0x000000018075cce0 __CFRunLoopDoSource0 + 208 (CFRunLoop.c:2016)
65  CoreFoundation                	0x0000000180696fe8 __CFRunLoopDoSources0 + 268 (CFRunLoop.c:2053)
66  CoreFoundation                	0x000000018069c7f4 __CFRunLoopRun + 820 (CFRunLoop.c:2951)
67  CoreFoundation                	0x00000001806b03b8 CFRunLoopRunSpecific + 600 (CFRunLoop.c:3268)
68  GraphicsServices              	0x000000019c68838c GSEventRunModal + 164 (GSEvent.c:2200)
69  UIKitCore                     	0x000000018304d0b8 -[UIApplication _run] + 1100 (UIApplication.m:3494)
70  UIKitCore                     	0x0000000182dcb1e8 UIApplicationMain + 2124 (UIApplication.m:5050)
71  App                         	0x0000000104e2eef0 main + 176 (main.swift:3)
72  dyld                          	0x0000000105ae5a24 start + 520 (dyldMain.cpp:876)
```

**Comment by morganchen12** [comment]
@foxware00 are you able to get the exception code of the crash? `mach_msg_trap` is a normal procedure when waiting for an xpc call to return. The `CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD` symbol just means your app is waiting on an external process (in this case cfprefsd), which Analytics uses to store consent settings.

Usually this fetch happens very quickly, but I'm wondering if it's blocking your app's main thread at launch and causing watchdog timer terminations.

**Comment by foxware00** [comment]
@morganchen12 the strange thing is Thread 0 isn't crashing, but another thread is. It's my code, but the line of the error doesn't make sense (it doesn't refer to an actual line of code). I'm looking at that, what is strange is. The Thread shown (main) above 9/10 looks like it's blocked on `CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD` in all the crash reports.

I appreciate this is a bit of a stab in the dark

**Comment by morganchen12** [comment]
Can you share the stack trace of all the threads?

**Comment by foxware00** [comment]
@morganchen12 here is an example of a crash

```
Incident Identifier: 5ECB2DC2-BBD5-41AE-85A7-346DAE417998
Hardware Model:      iPad13,4
Process:             App [703]
Path:                /private/var/containers/Bundle/Application/F950CAAB-0EB9-4BF8-A0DB-245AD49AF6A6/App.app/App
Identifier:          com.companyname.messaging.ios
Version:             3.62.0 (905)
AppStoreTools:       13A227
AppVariant:          1:iPad13,4:15
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Non UI
Parent Process:      launchd [1]
Coalition:           com.companyname.messaging.ios [732]

Date/Time:           2021-09-26 19:13:40.1111 +0100
Launch Time:         2021-09-26 18:58:43.4571 +0100
OS Version:          iPhone OS 15.1 (19B5042h)
Release Type:        Beta
Report Version:      104

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x0000000105d142d4
Exception Note:  EXC_CORPSE_NOTIFY
Terminating Process: exc handler [703]
Triggered by Thread:  5


Thread 0 name:
Thread 0:
0   libsystem_kernel.dylib        	0x00000001b6f79564 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001b6f79bfc mach_msg + 76 (mach_msg.c:119)
2   libdispatch.dylib             	0x00000001803b728c _dispatch_mach_send_and_wait_for_reply + 520 (mach.c:815)
3   libdispatch.dylib             	0x00000001803b763c dispatch_mach_send_with_result_and_wait_for_reply + 56 (mach.c:2019)
4   libxpc.dylib                  	0x00000001f048ab9c xpc_connection_send_message_with_reply_sync + 240 (connection.c:974)
5   CoreFoundation                	0x0000000180759c20 __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke_2 + 40 (CFPrefsSearchListSource.m:1025)
6   CoreFoundation                	0x00000001807266f8 -[_CFXPreferences withConnectionForRole:performBlock:] + 52 (CFXPreferences.m:1428)
7   CoreFoundation                	0x000000018074941c __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke + 140 (CFPrefsSearchListSource.m:1024)
8   CoreFoundation                	0x0000000180761cb0 CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD + 100 (CFPrefsSearchListSource.m:947)
9   CoreFoundation                	0x00000001807804ec -[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:] + 332 (CFPrefsSearchListSource.m:1044)
10  CoreFoundation                	0x00000001806d7824 -[CFPrefsSearchListSource alreadylocked_generationCountFromListOfSources:count:] + 232 (CFPrefsSearchListSource.m:1077)
11  CoreFoundation                	0x00000001806c6d04 -[CFPrefsSearchListSource alreadylocked_getDictionary:] + 468 (CFPrefsSearchListSource.m:1175)
12  CoreFoundation                	0x000000018072755c -[CFPrefsSearchListSource alreadylocked_copyValueForKey:] + 172 (CFPrefsSearchListSource.m:614)
13  CoreFoundation                	0x000000018078e958 -[CFPrefsSource copyValueForKey:] + 60 (CFPrefsSource.m:876)
14  CoreFoundation                	0x00000001806c8d7c __76-[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:]_block_invoke + 44 (CFXPreferences.m:1014)
15  CoreFoundation                	0x00000001806e8308 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 404 (CFPrefsSearchListSource.m:1728)
16  CoreFoundation                	0x00000001806c0328 normalizeQuintuplet + 356 (CFPrefsSearchListSource.m:74)
17  CoreFoundation                	0x00000001806b93ac -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 152 (CFPrefsSearchListSource.m:1600)
18  CoreFoundation                	0x00000001807170ec -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 168 (CFXPreferences.m:1010)
19  CoreFoundation                	0x00000001806dd4d8 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 128 (CFXPreferences.m:1896)
20  AppKit                      	0x00000001068b39bc 0x105b6c000 + 13924796
21  AppKit                      	0x00000001068109d0 +[APMAnalytics persistedConsentSource] + 40
22  AppKit                      	0x0000000106810814 +[APMAnalytics loadPersistedConsentSettings] + 32
23  AppKit                      	0x000000010680f690 __46+[APMAnalytics startWithAppID:origin:options:]_block_invoke + 80
24  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
25  libdispatch.dylib             	0x000000018039dee4 _dispatch_once_callout + 32 (once.c:52)
26  AppKit                      	0x000000010680f634 +[APMAnalytics startWithAppID:origin:options:] + 1048
27  AppKit                      	0x000000010680a230 +[FIRAnalytics startWithConfiguration:options:] + 532
28  AppKit                      	0x00000001062036f4 -[FIRApp configureCore] + 256 (FIRApp.m:368)
29  AppKit                      	0x00000001062033e0 +[FIRApp addAppToAppDictionary:] + 100 (FIRApp.m:324)
30  AppKit                      	0x0000000106202b7c +[FIRApp configureWithName:options:] + 704 (FIRApp.m:208)
31  AppKit                      	0x00000001062027f8 +[FIRApp configureWithOptions:] + 96 (FIRApp.m:151)
32  AppKit                      	0x0000000106202784 +[FIRApp configure] + 124 (FIRApp.m:137)
33  App                         	0x0000000104e60648 specialized AppDelegate.application(_:didFinishLaunchingWithOptions:) + 436 (AppDelegate.swift:41)
34  App                         	0x0000000104e591a8 application + 4 (<compiler-generated>:0)
35  App                         	0x0000000104e591a8 @objc AppDelegate.application(_:didFinishLaunchingWithOptions:) + 132
36  UIKitCore                     	0x0000000182e5c1ac -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 356 (UIApplication.m:2329)
37  UIKitCore                     	0x0000000183044548 -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 3524 (UIApplication.m:2741)
38  UIKitCore                     	0x000000018302ceb4 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1208 (UIApplication.m:4474)
39  UIKitCore                     	0x0000000182e8a054 -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 152 (_UISceneLifecycleMultiplexer.m:452)
40  UIKitCore                     	0x0000000182ce16a4 _UIScenePerformActionsWithLifecycleActionMask + 104 (_UISceneLifecycleState.m:109)
41  UIKitCore                     	0x0000000182e0db78 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 224 (_UISceneLifecycleMultiplexer.m:565)
42  UIKitCore                     	0x0000000182f2e644 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 380 (_UISceneLifecycleMultiplexer.m:517)
43  UIKitCore                     	0x000000018316f7bc -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 760 (_UISceneLifecycleMultiplexer.m:564)
44  UIKitCore                     	0x00000001830faed4 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 340 (_UISceneLifecycleMultiplexer.m:470)
45  UIKitCore                     	0x0000000182ce24a8 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block... + 196 (_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction.m:73)
46  UIKitCore                     	0x0000000182da2c88 +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:actions:completion:] + 892 (BSAnimationSettings+UIKit.m:50)
47  UIKitCore                     	0x0000000182ce43b8 _UISceneSettingsDiffActionPerformChangesWithTransitionContext + 276 (_UISceneSettingsDiffAction.m:23)
48  UIKitCore                     	0x0000000182dcf7e8 -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 384 (_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction.m:58)
49  UIKitCore                     	0x00000001832577bc __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke.610 + 776 (UIScene.m:1775)
50  UIKitCore                     	0x0000000182d131a4 -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 256 (UIScene.m:1476)
51  UIKitCore                     	0x0000000182d5693c -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 288 (UIScene.m:1742)
52  UIKitCore                     	0x0000000182e8ef18 -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 572 (UIApplication.m:3937)
53  UIKitCore                     	0x0000000182dcdaa8 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 388 (UIApplicationSceneClientAgent.m:45)
54  FrontBoardServices            	0x00000001927afe30 -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 440 (FBSScene.m:439)
55  FrontBoardServices            	0x00000001927d5cec __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke.215 + 128 (FBSWorkspaceScenesClient.m:384)
56  FrontBoardServices            	0x00000001927906c4 -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240 (FBSWorkspace.m:352)
57  FrontBoardServices            	0x0000000192791d04 __94-[FBSWorkspaceScenesClient createWithSceneID:groupID:parameters:transitionContext:completion:]_block_invoke + 372 (FBSWorkspaceScenesClient.m:383)
58  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
59  libdispatch.dylib             	0x00000001803a00f4 _dispatch_block_invoke_direct + 264 (queue.c:489)
60  FrontBoardServices            	0x0000000192791fa4 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48 (FBSSerialQueue.m:157)
61  FrontBoardServices            	0x00000001927913e4 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 220 (FBSSerialQueue.m:181)
62  FrontBoardServices            	0x00000001927959f4 -[FBSSerialQueue _performNextFromRunLoopSource] + 28 (FBSSerialQueue.m:194)
63  CoreFoundation                	0x000000018074c020 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28 (CFRunLoop.c:1972)
64  CoreFoundation                	0x000000018075cce0 __CFRunLoopDoSource0 + 208 (CFRunLoop.c:2016)
65  CoreFoundation                	0x0000000180696fe8 __CFRunLoopDoSources0 + 268 (CFRunLoop.c:2053)
66  CoreFoundation                	0x000000018069c7f4 __CFRunLoopRun + 820 (CFRunLoop.c:2951)
67  CoreFoundation                	0x00000001806b03b8 CFRunLoopRunSpecific + 600 (CFRunLoop.c:3268)
68  GraphicsServices              	0x000000019c68838c GSEventRunModal + 164 (GSEvent.c:2200)
69  UIKitCore                     	0x000000018304d0b8 -[UIApplication _run] + 1100 (UIApplication.m:3494)
70  UIKitCore                     	0x0000000182dcb1e8 UIApplicationMain + 2124 (UIApplication.m:5050)
71  App                         	0x0000000104e2eef0 main + 176 (main.swift:3)
72  dyld                          	0x0000000105ae5a24 start + 520 (dyldMain.cpp:876)

Thread 1:
0   libsystem_pthread.dylib       	0x00000001f0461e8c start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	0x00000001f0461e8c start_wqthread + 0

Thread 3 name:
Thread 3:
0   libsystem_kernel.dylib        	0x00000001b6f79564 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001b6f79bfc mach_msg + 76 (mach_msg.c:119)
2   libdispatch.dylib             	0x00000001803b728c _dispatch_mach_send_and_wait_for_reply + 520 (mach.c:815)
3   libdispatch.dylib             	0x00000001803b763c dispatch_mach_send_with_result_and_wait_for_reply + 56 (mach.c:2019)
4   libxpc.dylib                  	0x00000001f048ab9c xpc_connection_send_message_with_reply_sync + 240 (connection.c:974)
5   CoreFoundation                	0x0000000180759c20 __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke_2 + 40 (CFPrefsSearchListSource.m:1025)
6   CoreFoundation                	0x00000001807266f8 -[_CFXPreferences withConnectionForRole:performBlock:] + 52 (CFXPreferences.m:1428)
7   CoreFoundation                	0x000000018074941c __104-[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:]_block_invoke + 140 (CFPrefsSearchListSource.m:1024)
8   CoreFoundation                	0x0000000180761cb0 CFPREFERENCES_IS_WAITING_FOR_SYSTEM_CFPREFSD + 100 (CFPrefsSearchListSource.m:947)
9   CoreFoundation                	0x00000001807804ec -[CFPrefsSearchListSource synchronouslySendDaemonMessage:andAgentMessage:andDirectMessage:replyHandler:] + 332 (CFPrefsSearchListSource.m:1044)
10  CoreFoundation                	0x00000001806d7824 -[CFPrefsSearchListSource alreadylocked_generationCountFromListOfSources:count:] + 232 (CFPrefsSearchListSource.m:1077)
11  CoreFoundation                	0x00000001806c6d04 -[CFPrefsSearchListSource alreadylocked_getDictionary:] + 468 (CFPrefsSearchListSource.m:1175)
12  CoreFoundation                	0x000000018072755c -[CFPrefsSearchListSource alreadylocked_copyValueForKey:] + 172 (CFPrefsSearchListSource.m:614)
13  CoreFoundation                	0x000000018078e958 -[CFPrefsSource copyValueForKey:] + 60 (CFPrefsSource.m:876)
14  CoreFoundation                	0x00000001806c8d7c __76-[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:]_block_invoke + 44 (CFXPreferences.m:1014)
15  CoreFoundation                	0x00000001806e8308 __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 404 (CFPrefsSearchListSource.m:1728)
16  CoreFoundation                	0x00000001806c0328 normalizeQuintuplet + 356 (CFPrefsSearchListSource.m:74)
17  CoreFoundation                	0x00000001806b93ac -[_CFXPreferences withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 152 (CFPrefsSearchListSource.m:1600)
18  CoreFoundation                	0x00000001807170ec -[_CFXPreferences copyAppValueForKey:identifier:container:configurationURL:] + 168 (CFXPreferences.m:1010)
19  CoreFoundation                	0x00000001806dd4d8 _CFPreferencesCopyAppValueWithContainerAndConfiguration + 128 (CFXPreferences.m:1896)
20  AppKit                      	0x00000001068b39bc 0x105b6c000 + 13924796
21  AppKit                      	0x00000001068b3d68 0x105b6c000 + 13925736
22  AppKit                      	0x000000010686986c 0x105b6c000 + 13621356
23  AppKit                      	0x0000000106867784 0x105b6c000 + 13612932
24  libdispatch.dylib             	0x000000018039a8f0 _dispatch_call_block_and_release + 32 (init.c:1517)
25  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
26  libdispatch.dylib             	0x00000001803a3dc0 _dispatch_lane_serial_drain + 672 (inline_internal.h:2601)
27  libdispatch.dylib             	0x00000001803a4934 _dispatch_lane_invoke + 392 (queue.c:3937)
28  libdispatch.dylib             	0x00000001803af184 _dispatch_workloop_worker_thread + 656 (queue.c:6727)
29  libsystem_pthread.dylib       	0x00000001f04620f4 _pthread_wqthread + 288 (pthread.c:2541)
30  libsystem_pthread.dylib       	0x00000001f0461e94 start_wqthread + 8

Thread 4 name:
Thread 4:
0   libsystem_kernel.dylib        	0x00000001b6f79564 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001b6f79bfc mach_msg + 76 (mach_msg.c:119)
2   CoreFoundation                	0x0000000180698688 __CFRunLoopServiceMachPort + 372 (CFRunLoop.c:2646)
3   CoreFoundation                	0x000000018069c97c __CFRunLoopRun + 1212 (CFRunLoop.c:3000)
4   CoreFoundation                	0x00000001806b03b8 CFRunLoopRunSpecific + 600 (CFRunLoop.c:3268)
5   Foundation                    	0x0000000181ec9c74 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236 (NSRunLoop.m:373)
6   Foundation                    	0x0000000181f0b548 -[NSRunLoop(NSRunLoop) runUntilDate:] + 92 (NSRunLoop.m:420)
7   UIKitCore                     	0x0000000182fc6068 -[UIEventFetcher threadMain] + 524 (UIEventFetcher.m:1167)
8   Foundation                    	0x0000000181f1980c __NSThread__start__ + 808 (NSThread.m:972)
9   libsystem_pthread.dylib       	0x00000001f04629a4 _pthread_start + 148 (pthread.c:891)
10  libsystem_pthread.dylib       	0x00000001f0461ea0 thread_start + 8

Thread 5 name:
Thread 5 Crashed:
0   AppKit                      	0x0000000105d142d4 closure #1 in DataManager.markDbDirty(pollingId:) + 416 (DatabaseDelegate.swift:0)
1   AppKit                      	0x0000000105d141e4 closure #1 in DataManager.markDbDirty(pollingId:) + 176 (DataManager.swift:0)
2   AppKit                      	0x000000010659d25c partial apply for thunk for @escaping @callee_guaranteed () -> (@owned Observable<A.ObservableConvertibleType.Element>, @error @owned Error) + 24 (<compiler-generated>:0)
3   AppKit                      	0x000000010659d498 DeferredSink.run() + 184 (Deferred.swift:36)
4   AppKit                      	0x000000010659d81c Deferred.run<A>(_:cancel:) + 220 (Deferred.swift:72)
5   AppKit                      	0x00000001065c65b8 closure #1 in Producer.subscribe<A>(_:) + 228 (Producer.swift:26)
6   AppKit                      	0x0000000106598374 specialized CurrentThreadScheduler.schedule<A>(_:action:) + 212 (<compiler-generated>:0)
7   AppKit                      	0x00000001065c6224 Producer.subscribe<A>(_:) + 504 (Producer.swift:24)
8   AppKit                      	0x00000001065bd528 protocol witness for ObservableType.subscribe<A>(_:) in conformance Observable<A> + 28 (<compiler-generated>:0)
9   AppKit                      	0x00000001065e074c closure #1 in SubscribeOnSink.run() + 152 (SubscribeOn.swift:78)
10  AppKit                      	0x00000001065e492c thunk for @escaping @callee_guaranteed () -> (@out Disposable) + 20 (<compiler-generated>:0)
11  AppKit                      	0x00000001065a2af0 partial apply for closure #1 in DispatchQueueConfiguration.schedule<A>(_:action:) + 120 (DispatchQueueConfiguration.swift:27)
12  AppKit                      	0x0000000105bd2148 thunk for @escaping @callee_guaranteed () -> () + 20 (<compiler-generated>:0)
13  libdispatch.dylib             	0x000000018039a8f0 _dispatch_call_block_and_release + 32 (init.c:1517)
14  libdispatch.dylib             	0x000000018039c63c _dispatch_client_callout + 20 (object.m:560)
15  libdispatch.dylib             	0x00000001803a3dc0 _dispatch_lane_serial_drain + 672 (inline_internal.h:2601)
16  libdispatch.dylib             	0x00000001803a4934 _dispatch_lane_invoke + 392 (queue.c:3937)
17  libdispatch.dylib             	0x00000001803af184 _dispatch_workloop_worker_thread + 656 (queue.c:6727)
18  libsystem_pthread.dylib       	0x00000001f04620f4 _pthread_wqthread + 288 (pthread.c:2541)
19  libsystem_pthread.dylib       	0x00000001f0461e94 start_wqthread + 8

Thread 6 name:
Thread 6:
0   libsystem_kernel.dylib        	0x00000001b6f79564 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001b6f79bfc mach_msg + 76 (mach_msg.c:119)
2   CoreFoundation                	0x0000000180698688 __CFRunLoopServiceMachPort + 372 (CFRunLoop.c:2646)
3   CoreFoundation                	0x000000018069c97c __CFRunLoopRun + 1212 (CFRunLoop.c:3000)
4   CoreFoundation                	0x00000001806b03b8 CFRunLoopRunSpecific + 600 (CFRunLoop.c:3268)
5   CFNetwork                     	0x0000000181123860 +[__CFN_CoreSchedulingSetRunnable _run:] + 464 (CoreSchedulingSet.mm:1372)
6   Foundation                    	0x0000000181f1980c __NSThread__start__ + 808 (NSThread.m:972)
7   libsystem_pthread.dylib       	0x00000001f04629a4 _pthread_start + 148 (pthread.c:891)
8   libsystem_pthread.dylib       	0x00000001f0461ea0 thread_start + 8

Thread 7:
0   libsystem_pthread.dylib       	0x00000001f0461e8c start_wqthread + 0

Thread 8:
0   libsystem_pthread.dylib       	0x00000001f0461e8c start_wqthread + 0


Thread 5 crashed with ARM Thread State (64-bit):
    x0: 0x0000000282b901f0   x1: 0x000000016b31a1f8   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x0000000a00000003   x5: 0x000000000000000e   x6: 0x0000000800000003   x7: 0x000000000000000e
    x8: 0x00000001071057e0   x9: 0x0000000000000000  x10: 0x0000000000000000  x11: 0x0000000000000001
   x12: 0xbffffffe505338bc  x13: 0x00000000062ea000  x14: 0x0000000006200000  x15: 0x000000016b31a138
   x16: 0x0000000185445124  x17: 0x00000000000008fd  x18: 0x0000000000000000  x19: 0xe000000000000000
   x20: 0x0000000000000000  x21: 0x0000000000000000  x22: 0x0000000281894280  x23: 0x00000002829ac540
   x24: 0x00000002829ac550  x25: 0x0000000105a1cd40  x26: 0x0000000105a1d0f0  x27: 0x000000016b31a2c0
   x28: 0x000000016b31a2b0   fp: 0x000000016b31a280   lr: 0xc02f598105d141e4
    sp: 0x000000016b31a1e0   pc: 0x0000000105d142d4 cpsr: 0x20001000
   esr: 0xf2000001 (Breakpoint) brk 1


Binary Images:
0x104e28000 - 0x10551ffff App arm64  <788c0f0cdf063d62b25da8d8fbf02f9c> /private/var/containers/Bundle/Application/F950CAAB-0EB9-4BF8-A0DB-245AD49AF6A6/App.app/App
0x105acc000 - 0x105b23fff dyld arm64e  <13216777f4be3a7bb470857e80ad3300> /usr/lib/dyld
0x105b6c000 - 0x106b83fff AppKit arm64  <d8c830eccdb237428975121208bf9b19> /private/var/containers/Bundle/Application/F950CAAB-0EB9-4BF8-A0DB-245AD49AF6A6/App.app/Frameworks/AppKit.framework/AppKit
0x180398000 - 0x1803defff libdispatch.dylib arm64e  <4507091349373f3e9257f5e4a8333c4b> /usr/lib/system/libdispatch.dylib
0x180691000 - 0x180ae4fff CoreFoundation arm64e  <4744e09d102038dfb3571a8883e7a543> /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation
0x180ea9000 - 0x18136dfff CFNetwork arm64e  <b1178efce4833180babc6efe0d6db47b> /System/Library/Frameworks/CFNetwork.framework/CFNetwork
0x181eb1000 - 0x1821b5fff Foundation arm64e  <bc23c5ce147f3c639dea60880e0a60d7> /System/Library/Frameworks/Foundation.framework/Foundation
0x182b33000 - 0x1843b7fff UIKitCore arm64e  <e1ad82d4c7da31b6b13f90ce2b21328e> /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore
0x192786000 - 0x192831fff FrontBoardServices arm64e  <7461b11e23d63fe49a9f28a6a4676a58> /System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices
0x19c687000 - 0x19c68ffff GraphicsServices arm64e  <78133c6a27393664ab75318955b965d8> /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices
0x1b6f78000 - 0x1b6fabfff libsystem_kernel.dylib arm64e  <5af49e07b3313793958fffeb1773568a> /usr/lib/system/libsystem_kernel.dylib
0x1f0461000 - 0x1f046cfff libsystem_pthread.dylib arm64e  <2c38d28d4d813c45acfca908783e1c25> /usr/lib/system/libsystem_pthread.dylib
0x1f047c000 - 0x1f04b3fff libxpc.dylib arm64e  <69bee4416e5c358883d17776dd3a7460> /usr/lib/system/libxpc.dylib

EOF

```

**Comment by morganchen12** [comment]
The `EXC_BREAKPOINT` running on arm64 in Swift code usually is the result of a Swift fatal error, like a nil force-unwrap.

```
Thread 5 name:
Thread 5 Crashed:
0   AppKit                      	0x0000000105d142d4 closure #1 in DataManager.markDbDirty(pollingId:) + 416 (DatabaseDelegate.swift:0)
1   AppKit                      	0x0000000105d141e4 closure #1 in DataManager.markDbDirty(pollingId:) + 176 (DataManager.swift:0)
```

Can you share the source of the DataManager type?

**Comment by foxware00** [comment]
@morganchen12 unforuntately not. There isn't 416 lines in DataManager or DatabaseDelegate so not 100% sure what's happening. I'm happy to close this, but out of around 50 crashes 45 or so all have Thread 0 in the same state. I will refactor DataManager a little and see if the crashing persists
